### PR TITLE
Feature/add service endpoint

### PR DESCRIPTION
**Pull request by sheenacodes**
added endpoint service with tests

### CODE REVIEW

File: service-endpoint/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+USER app
+WORKDIR /home/app
+
+# Copy and install requirements only first to cache the dependency layer
+COPY --chown=app:app requirements.txt .
+RUN pip install --no-cache-dir --no-compile --upgrade -r requirements.txt
+
+COPY --chown=app:app . .
+
+CMD ["/home/app/.local/bin/uvicorn", "app:app", "--host", "0.0.0.0", "--port", "8000", "--proxy-headers"]
```
Comment (aapris) [inline service-endpoint/Dockerfile:18]: We should have `--reload` here when developing @laurigates 
https://www.uvicorn.org/settings/#development
Would it be easy to add it, when we are running this locally?

File: service-endpoint/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+dynamic = ["version"]
+
+dependencies = [
+  # "aiohttp",
+  "aiokafka",
+  # "aioredis",
+  "fastapi",
+  "fvhiot@https://github.com/ForumViriumHelsinki/FVHIoT-python/archive/refs/heads/master.zip",
+  "kafka-python",
+  # "msgpack",
```
Comment (aapris) [inline service-endpoint/pyproject.toml:18]: Is this functional without msgpack? data_pack() function uses it to serialise http request data. Also aioredis, aiohttp and xmltodict should be removed, if they are not needed @sheenacodes .

**Review by aapris — CHANGES_REQUESTED** [review]
Check the comments and fix those which can be fixed easily. Then merge.

File: service-endpoint/pyproject.toml (hunk 2 above)
Comment (sheenacodes) [reply service-endpoint/pyproject.toml]: rremoved commented line.
The service is functional as per current tests without msgpack

File: service-endpoint/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+readme = "README.md"
+requires-python = ">=3.9"
+dynamic = ["version"]
+
+dependencies = [
+  # "aiohttp",
+  "aiokafka",
+  # "aioredis",
+  "fastapi",
+  "fvhiot@https://github.com/ForumViriumHelsinki/FVHIoT-python/archive/refs/heads/master.zip",
```
[the remark this answers is not in the document]
Comment (laurigates) [reply service-endpoint/pyproject.toml:14]: Yes, that would be a lot better. At the moment the URL for the latest release tag seems to be `https://github.com/ForumViriumHelsinki/FVHIoT-python/archive/refs/tags/v0.3.0.zip`.

An even cleaner/cooler approach would be to have GitHub actions create packages when releases are tagged in the FVHIoT repo. I can have a look at getting that done, but for now using the link to the release source zip probably works just as well.

File: service-endpoint/Dockerfile (hunk 1 above)
Comment (laurigates) [reply service-endpoint/Dockerfile:18]: This should be possible by setting `UVICORN_RELOAD=True` in docker-compose.yml.

**Pull request merged**